content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,615 @@
+- Showing how much XRP is set aside for the account's [reserve requirement](reserves.html).
+- Sending [direct XRP payments](direct-xrp-payments.html), and providing feedback about the intended destination address, including:
+    - Whether the intended destination already exists in the XRP Ledger, or the payment would have to fund its creation.
+    - If the address doesn't want to receive XRP ([`DisallowXRP` flag](become-an-xrp-ledger-gateway.html#disallow-xrp) enabled).
+    - If the address has a [verified domain name](https://xrpl.org/xrp-ledger-toml.html#account-verification) associated with it.
+
+The application in this tutorial _doesn't_ have the ability to send or trade [tokens](issued-currencies.html) or 
+use other [payment types](payment-types.html) like [Escrow](https://xrpl.org/escrow.html) or [Payment Channels](https://xrpl.org/payment-channels.html). However, it provides a foundation 
+that you can implement those and other features on top of.
+
+In addition to the above features, you'll also learn a bit about Events, IPC (inter-process-communication) 
+and asynchronous (async) code in JavaScript.
+
+## Steps
+
+### 0. Project setup
+
+To initialize the project we will create a package.json file with the following content:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/package.json", language="js", lines="1-28") }}
+
+Here we define the libraries our application will use in the `dependencies` section as well as shortcuts for running our application in the `scripts` section. To install those dependencies, run the following command:
+```console
+npm install
+```
+
+This installs the Electron Framework, the xrpl.js client library and a couple of helpers we are going to need for our
+application to work. 
+
+### 1. Hello XRP Ledger
+
+**Full code for this step:** 
+[`1_hello.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/1_hello.js),
+[`view/1_preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_preload.js),
+[`view/1_hello.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_hello.html),
+[`view/1_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/1_renderer.js).
+
+Our first step is to write a "Hello World" like application, that interacts on a very basic level with the XRP Ledger 
+and displays some high-level information about the current ledger state on the screen. Nothing too fancy so far, as we 
+will take care of styling and GUI related coding in a later step:
+
+![Screenshot: Step 1, hello world equivalent](img/javascript-wallet-1.png)
+
+First, we will create an entrypoint for our application, for this we create the file `1_hello.js` with the following content:
+
+`1_hello.js`
```

Comment:
I was a bit confused at this point because I thought that the "Hello World" application would incldue "Hello World" somewhere - (e.g. just proving that we can set up the desktop view correctly) 

Maybe it'd be better to call it "Basic view of the ledger" or to actually include a "Hello World" screen before any ledger work to isolate getting Electron working? (Not a big deal either way - just thought I'd flag)

Reply:
This would also let you split up the part that creates the application window, and the part that calls the XRP Ledger's API into two steps so each can get their own explanation.

Reply:
@JST5000, We had this issue of flip-flopping from "we" to "you" before, this was because of the code-along style of the  original python tutorial. As this one incorporates more code, with electron and the extra password step, somehow that explanatory style with "now we have" crept in. I guess I will reformat the description to stick to the code-along style, including descriptions where necessary. then all three wallet tutorials conform to the same style.

This will include working on just one file (e.g. preload.js instead of 38_preload.js) without the numbers, which will be used only for reference. While we're on it, look if theres a dusty box of medals somewhere in the basement, this one turned out way more complex than expected :)

Reply:
This has been added to Step 0 / Setup which is now a real "Hello world!" application, Step 2 is now aptly called "Ledger Index"